### PR TITLE
Documentation: Add how-to for building a Debug binary

### PR DESCRIPTION
**Pull request by sideshowbarker**

### CODE REVIEW

**Comment by sideshowbarker** [comment]
> This disables optimizations right?

Yes, as far as I understand, it does. Though per @trflynn89 in https://discord.com/channels/1247070541085671459/1247090705013280769/1260224384207949905, it:
> _also puts the build in a different subfolder under `Build/` so that you don't have to rebuild when you go back and forth between debug and default_

So from that I take it the build’s intentionally designed to allow you to do a debug build without clobbering your existing “default”/release build (with the optimizations).

All that said, it doesn’t completely match what I actually see in the `Build/` in my own environment, which is this:

```
$ ls Build/
drwxr-xr-x@  4 mike   128B Jul  8 20:12 caches
drwxr-xr-x   6 mike   192B Jul 10 06:58 ladbyird
drwxr-xr-x@ 21 mike   672B Jul 10 07:56 ladybird
```
I do see two subfolders in there: the normal, expected `ladybird` one, plus an additional `ladbyird` one. But as far as the contents of that `ladbyird` one:

```
$ ls Build/ladbyird/
-rw-r--r--  1 mike    28K Jul  9 22:02 CMakeCache.txt
drwxr-xr-x  6 mike   192B Jul 10 06:58 CMakeFiles
drwxr-xr-x  7 mike   224B Jul  9 22:02 Lagom
drwxr-xr-x  3 mike    96B Jul  9 22:02 caches
```
That is, there’s no separate `bin` directory in there, and so no separate `Ladybird.app`.

And the `Build/ladybird/bin/Ladybird.app` (in the normal build subdirectory) is the _debug_ build.

In other words, in my environment at least, it does seem like the debug build clobbered my default/release build — and I don’t have any way to switch back and forth between the two.

So I must not yet be doing it completely right, and maybe Tim and weigh in here with some more guidance.

**Comment by ADKaster** [comment]
When running ./Meta/serenity.sh, it doesn't know about the expected `binaryDir` difference. I believe when it does cmake --preset Debug, it also passes the binary directory explicitly, which is what causes the clobbering issue.

**Comment by ADKaster** [comment]
@sideshowbarker #578 should resolve the build directory problem
